In [3]:
print("Nespresso CRM Analysis Project")

Nespresso CRM Analysis Project


In [4]:
import pandas as pd
data = pd.read_excel("RD.xlsx")
data

Invoice Type INVOICENUMBER INVOICEDATE Channel    Item Group  \
0          Invoice   EINC-152524  2023-01-01     CRC  B2C Capsules   
1          Invoice   EINC-152524  2023-01-01     CRC  B2C Capsules   
2          Invoice   EINC-152524  2023-01-01     CRC  B2C Capsules   
3          Invoice   EINC-152524  2023-01-01     CRC  B2C Capsules   
4          Invoice   EINC-152524  2023-01-01     CRC  B2C Capsules   
...            ...           ...         ...     ...           ...   
73587      Invoice   EINC-304454  2024-10-31     CRC  B2C Capsules   
73588      Invoice   EINC-304481  2024-10-31     CRC  B2C Capsules   
73589      Invoice   EINC-304481  2024-10-31     CRC  B2C Capsules   
73590      Invoice   EINC-304481  2024-10-31     CRC  B2C Capsules   
73591      Invoice   EINC-304453  2024-10-31     CRC  B2C Capsules   

       PRODUCTNUMBER                PRODUCTNAME  QUANTITY CUSTOMERACCOUNT  
0             773100      BC CCA TRF كوكو ترافل        30      1100305515  
1             760800             Paris Espresso        30      1100305515  
2             756800   Ristretto Italy ريستريتو       180      1100305515  
3             772800  BC Van Ecler فانيلا اكلير        30      1100305515  
4             776400             Freddo Intenso        30      1100305515  
...              ...                        ...       ...             ...  
73587         776500    ISP ARPEGGIO 20x53g R70       120      2100600447  
73588         776500    ISP ARPEGGIO 20x53g R70       120      1100905041  
73589         776500    ISP ARPEGGIO 20x53g R70        60      1100905041  
73590         774600  ISP RIST DECAF 20x57g R70        60      1100905041  
73591         776500    ISP ARPEGGIO 20x53g R70        30        C-046867  

[73592 rows x 9 columns]

In [5]:
print("Calculate Total Active Customers")
print("The Total Active Customers In 2023 & 2024 IS:",len(data["CUSTOMERACCOUNT"].unique()))

Calculate Total Active Customers
The Total Active Customers In 2023 & 2024 IS: 11454


In [6]:
print("Calculate Active Customers In 2023")
print("The Total Active Customers In 2023 Is:",
len(data[(data["INVOICEDATE"]>="2023-1-1") & (data["INVOICEDATE"]<="2023-12-31")]["CUSTOMERACCOUNT"].unique()))

Calculate Active Customers In 2023
The Total Active Customers In 2023 Is: 8612


In [7]:
print("Calculate Active Customers In 2024")
print("The Total Active Customers In 2024 Is:",
len(data[(data["INVOICEDATE"]>="2024-1-1") & (data["INVOICEDATE"]<="2024-12-31")]["CUSTOMERACCOUNT"].unique()))

Calculate Active Customers In 2024
The Total Active Customers In 2024 Is: 4247


In [8]:
data[(data["INVOICEDATE"]>="2024-1-1") & (data["INVOICEDATE"]<="2024-12-31")]["CUSTOMERACCOUNT"].unique()

array(['1100201438', '1100215113', '1100220303', ..., '2100600447',
       '1100905041', 'C-046867'], dtype=object)

In [9]:
data[(data["INVOICEDATE"]>="2023-1-1") & (data["INVOICEDATE"]<="2023-12-31")]["CUSTOMERACCOUNT"].unique()

array(['1100305515', '1100912816', '1100903570', ..., '1100215113',
       '1100900496', 'C-051797'], dtype=object)

In [10]:
i = 0
for CustomerAccount in data[(data["INVOICEDATE"]>="2024-1-1") & (data["INVOICEDATE"]<="2024-12-31")]["CUSTOMERACCOUNT"].unique():
    if CustomerAccount in data[(data["INVOICEDATE"]>="2023-1-1") & (data["INVOICEDATE"]<="2023-12-31")]["CUSTOMERACCOUNT"].unique():
        i = i + 1
print(i)

1405


In [11]:
i = 0
for CustomerAccount in data[(data["INVOICEDATE"]>="2024-1-1") & (data["INVOICEDATE"]<="2024-12-31")]["CUSTOMERACCOUNT"].unique():
    if CustomerAccount not in data[(data["INVOICEDATE"]>="2023-1-1") & (data["INVOICEDATE"]<="2023-12-31")]["CUSTOMERACCOUNT"].unique():
        i = i + 1
print(i)

2842


In [12]:
print(1405 + 2842)

4247


In [13]:
print("New Customers In 2024:",2842)
print("Exicting Customers In 2024:",1405)

New Customers In 2024: 2842
Exicting Customers In 2024: 1405


In [14]:
print("Attrition Rate")
print("Attrition Rate = ",round(((8612 - 4247) / 8612) * 100,2),"%")

Attrition Rate
Attrition Rate =  50.69 %


In [15]:
print("Sold Capsules 2023 & 2024")
print("Total Sold Capsules In 2023 & 2024 IS:",sum(data["QUANTITY"]))

Sold Capsules 2023 & 2024
Total Sold Capsules In 2023 & 2024 IS: 5614653


In [16]:
print("Sold Capsules In 2023 IS:",
sum(data[(data["INVOICEDATE"]>="2023-1-1") & (data["INVOICEDATE"]<="2023-12-31")]["QUANTITY"]))

Sold Capsules In 2023 IS: 4124253


In [17]:
print("Sold Capsules In 2024 IS:",
sum(data[(data["INVOICEDATE"]>="2024-1-1") & (data["INVOICEDATE"]<="2024-12-31")]["QUANTITY"]))

Sold Capsules In 2024 IS: 1490400


In [18]:
print("Number of Orders 2023 & 2024")
print("Total Sold Capsules In 2023 & 2024 IS:",data["INVOICENUMBER"].nunique())

Number of Orders 2023 & 2024
Total Sold Capsules In 2023 & 2024 IS: 20816


In [19]:
print("Number of Orders In 2023 IS:",
data[(data["INVOICEDATE"]>="2023-1-1") & (data["INVOICEDATE"]<="2023-12-31")]["INVOICENUMBER"].nunique())

Number of Orders In 2023 IS: 14188


In [20]:
print("Number of Orders In 2024 IS:",
data[(data["INVOICEDATE"]>="2024-1-1") & (data["INVOICEDATE"]<="2024-12-31")]["INVOICENUMBER"].nunique())

Number of Orders In 2024 IS: 6628


In [21]:
print("Average Order Size in Caps 2023 & 2024:",
sum(data["QUANTITY"]) / data["INVOICENUMBER"].nunique())

Average Order Size in Caps 2023 & 2024: 269.7277574942352


In [22]:
print("Average Order Size in Caps 2023 IS:", 4124253 / 14188)

Average Order Size in Caps 2023 IS: 290.6860022554271


In [23]:
print("Average Order Size in Caps 2024 IS:", 1490400 / 6628)

Average Order Size in Caps 2024 IS: 224.86421243210623


In [24]:
print("****************************************************")

****************************************************


In [25]:
#data

In [26]:
DataFrame = pd.DataFrame(data)

In [27]:
DataFrame

Invoice Type INVOICENUMBER INVOICEDATE Channel    Item Group  \
0          Invoice   EINC-152524  2023-01-01     CRC  B2C Capsules   
1          Invoice   EINC-152524  2023-01-01     CRC  B2C Capsules   
2          Invoice   EINC-152524  2023-01-01     CRC  B2C Capsules   
3          Invoice   EINC-152524  2023-01-01     CRC  B2C Capsules   
4          Invoice   EINC-152524  2023-01-01     CRC  B2C Capsules   
...            ...           ...         ...     ...           ...   
73587      Invoice   EINC-304454  2024-10-31     CRC  B2C Capsules   
73588      Invoice   EINC-304481  2024-10-31     CRC  B2C Capsules   
73589      Invoice   EINC-304481  2024-10-31     CRC  B2C Capsules   
73590      Invoice   EINC-304481  2024-10-31     CRC  B2C Capsules   
73591      Invoice   EINC-304453  2024-10-31     CRC  B2C Capsules   

       PRODUCTNUMBER                PRODUCTNAME  QUANTITY CUSTOMERACCOUNT  
0             773100      BC CCA TRF كوكو ترافل        30      1100305515  
1             760800             Paris Espresso        30      1100305515  
2             756800   Ristretto Italy ريستريتو       180      1100305515  
3             772800  BC Van Ecler فانيلا اكلير        30      1100305515  
4             776400             Freddo Intenso        30      1100305515  
...              ...                        ...       ...             ...  
73587         776500    ISP ARPEGGIO 20x53g R70       120      2100600447  
73588         776500    ISP ARPEGGIO 20x53g R70       120      1100905041  
73589         776500    ISP ARPEGGIO 20x53g R70        60      1100905041  
73590         774600  ISP RIST DECAF 20x57g R70        60      1100905041  
73591         776500    ISP ARPEGGIO 20x53g R70        30        C-046867  

[73592 rows x 9 columns]

In [28]:

groupbyCustomer23 = DataFrame[(data["INVOICEDATE"]>="2023-1-1") & (data["INVOICEDATE"]<="2023-12-31")].groupby("CUSTOMERACCOUNT")["QUANTITY"].sum()

In [29]:
groupbyCustomer23

CUSTOMERACCOUNT
220022014      450
1000000112    1110
1000000694     750
1000000704    2100
1000000705     120
              ... 
C-051948        90
C-052001        60
C-052050      1500
C-052056        60
C-052073        60
Name: QUANTITY, Length: 8612, dtype: int64

In [30]:
Ultimate = 0
Intensive = 0
Regular = 0
Moderate = 0
for quantity in groupbyCustomer23:
    if quantity >= 1921:
        Ultimate = Ultimate + 1
    elif quantity >= 721:
        Intensive = Intensive + 1
    elif quantity >= 361:
        Regular = Regular + 1
    else:
        Moderate = Moderate + 1
print("Base.")
print("Ultimate Base =", Ultimate)
print("Intensive Base =", Intensive)
print("Regular Base =", Regular)
print("Moderate Base =", Moderate)

Base.
Ultimate Base = 340
Intensive Base = 973
Regular Base = 1298
Moderate Base = 6001


In [31]:
print("Sum Of All Segments Base:",Ultimate + Intensive + Regular + Moderate)
TotalBase = Ultimate + Intensive + Regular + Moderate

Sum Of All Segments Base: 8612


In [32]:
print("Weight.")
print("Ultimate Weight =", round((Ultimate / TotalBase) * 100,2))
print("Intensive Weight =", round((Intensive / TotalBase) * 100,2))
print("Regular Weight =", round((Regular / TotalBase) * 100,2))
print("Moderate Weight =", round((Moderate / TotalBase) * 100,2))

Weight.
Ultimate Weight = 3.95
Intensive Weight = 11.3
Regular Weight = 15.07
Moderate Weight = 69.68


In [33]:
groupbyCustomer24 = DataFrame[(data["INVOICEDATE"]>="2024-1-1") & (data["INVOICEDATE"]<="2024-12-31")].groupby("CUSTOMERACCOUNT")["QUANTITY"].sum()

In [34]:
groupbyCustomer24

CUSTOMERACCOUNT
1000000065    120
1000000320    240
1000000462    390
1000000481    180
1000000512     90
             ... 
C-058631      180
C-058641      180
C-058657      390
C-058664      390
C-058695      390
Name: QUANTITY, Length: 4247, dtype: int64

In [35]:
Ultimate = 0
Intensive = 0
Regular = 0
Moderate = 0
for quantity in groupbyCustomer24:
    if quantity >= 1921:
        Ultimate = Ultimate + 1
    elif quantity >= 721:
        Intensive = Intensive + 1
    elif quantity >= 361:
        Regular = Regular + 1
    else:
        Moderate = Moderate + 1
print("Base.")
print("Ultimate Base =", Ultimate)
print("Intensive Base =", Intensive)
print("Regular Base =", Regular)
print("Moderate Base =", Moderate)

Base.
Ultimate Base = 141
Intensive Base = 307
Regular Base = 469
Moderate Base = 3330


In [36]:
print("Sum Of All Segments Base:",Ultimate + Intensive + Regular + Moderate)
TotalBase = Ultimate + Intensive + Regular + Moderate

Sum Of All Segments Base: 4247


In [37]:
print("Weight.")
print("Ultimate Weight =", round((Ultimate / TotalBase) * 100,2))
print("Intensive Weight =", round((Intensive / TotalBase) * 100,2))
print("Regular Weight =", round((Regular / TotalBase) * 100,2))
print("Moderate Weight =", round((Moderate / TotalBase) * 100,2))

Weight.
Ultimate Weight = 3.32
Intensive Weight = 7.23
Regular Weight = 11.04
Moderate Weight = 78.41


In [38]:
Ultimate = 0
Intensive = 0
Regular = 0
Moderate = 0
for quantity in groupbyCustomer24:
    if quantity >= 1921:
        Ultimate = Ultimate + quantity
    elif quantity >= 721:
        Intensive = Intensive + quantity
    elif quantity >= 361:
        Regular = Regular + quantity
    else:
        Moderate = Moderate + quantity
print("Base.")
print("Ultimate Capsules =", Ultimate)
print("Intensive Capsules =", Intensive)
print("Regular Capsules =", Regular)
print("Moderate Capsules =", Moderate)

Base.
Ultimate Capsules = 553230
Intensive Capsules = 355290
Regular Capsules = 250290
Moderate Capsules = 331590


In [39]:
print("Sum Of All Segments Capsules:",Ultimate + Intensive + Regular + Moderate)
TotalBase = Ultimate + Intensive + Regular + Moderate

Sum Of All Segments Capsules: 1490400


In [40]:
print("Weight.")
print("Ultimate Weight =", round((Ultimate / TotalBase) * 100,2))
print("Intensive Weight =", round((Intensive / TotalBase) * 100,2))
print("Regular Weight =", round((Regular / TotalBase) * 100,2))
print("Moderate Weight =", round((Moderate / TotalBase) * 100,2))

Weight.
Ultimate Weight = 37.12
Intensive Weight = 23.84
Regular Weight = 16.79
Moderate Weight = 22.25


In [41]:
print("********************************************************************************")

********************************************************************************


In [42]:
print("The Summary Of Analysis")

The Summary Of Analysis


# The Total Active Customers In 2023 Is: 8612.
# The Total Active Customers In 2024 Is: 4247.
# New Customers In 2024 IS: 2842.
# Exicting Customers In 2024 IS: 1405
# Attrition Rate In 2024 IS:  50.69 %
# Sold Capsules In 2023 IS: 4124253
# Sold Capsules In 2024 IS: 1490400
# Number of Orders In 2023 IS: 14188
# Number of Orders In 2024 IS: 6628
# Average Order Size in Caps 2023 IS: 290.68
# Average Order Size in Caps 2024 IS: 224.86

# Please fill in number of customers in every segment			

# 2023

# Base.
# Ultimate Base = 340
# Intensive Base = 973
# Regular Base = 1298
# Moderate Base = 6001

# Sum Of All Segments Base: 8612

# Weight.
# Ultimate Weight = 3.95
# Intensive Weight = 11.3
# Regular Weight = 15.07
# Moderate Weight = 69.68

# 2024

# Base.
# Ultimate Base = 141
# Intensive Base = 307
# Regular Base = 469
# Moderate Base = 3330

# Sum Of All Segments Base: 4247

# Weight.
# Ultimate Weight = 3.32
# Intensive Weight = 7.23
# Regular Weight = 11.04
# Moderate Weight = 78.41

# 2024

# Base.
# Ultimate Capsules = 553230
# Intensive Capsules = 355290
# Regular Capsules = 250290
# Moderate Capsules = 331590

# Sum Of All Segments Base: 1490400

# Weight.
# Ultimate Weight = 37.12
# Intensive Weight = 23.84
# Regular Weight = 16.79
# Moderate Weight = 22.25

# >>>>>>>>>>>>>>>>>The End Thank You<<<<<<<<<<<<<<<<<